webrtc/videoroom/lib/videoroom_web/room_channel.ex

```diff
@@ -97,19 +102,44 @@ defmodule VideoRoomWeb.RoomChannel do
     {:noreply, socket}
   end
 
-  def handle_info({:signal, {:sdp_offer, sdp}, participants}, socket) do
+  def handle_info({:signal, {:sdp_offer, sdp}, participants, user_id}, socket) do
     participants =
       Enum.map(
         participants,
         &%{
+          "id" => &1.id,
           "displayName" => &1.display_name,
           "mids" => &1.mids,
           "mutedAudio" => &1.muted_audio,
           "mutedVideo" => &1.muted_video
         }
       )
 
-    push(socket, "offer", %{data: %{"type" => "offer", "sdp" => sdp}, participants: participants})
+    push(socket, "offer", %{
+      data: %{"type" => "offer", "sdp" => sdp},
+      participants: participants,
+      userId: user_id
+    })
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_info({:participants_list, participants}, socket) do
+    participants =
+      Enum.map(
+        participants,
+        &%{
+          "id" => &1.id,
+          "displayName" => &1.display_name,
+          "mids" => &1.mids,
+          "mutedAudio" => &1.muted_audio,
+          "mutedVideo" => &1.muted_video
+        }
+      )
```

Comment:
This is duplicated with lines 106-116 in `handle_info({:signal, {:sdp_offer, sdp}...` Also we have `get_participant_data` function that seems to prepare data to send but here we do almost the same thing by replacing keys to strings. Shouldn't we do this in get_participant_data?

Reply:
I think, that making such a preparation is dependency, that we don't want to have in pipeline.ex, where get_participants_data is placed, so I extracted functionality of this piece of code to separate, private function and made it more generic